automated install/uninstall.sh

```diff
@@ -83,11 +83,11 @@ removePiholeFiles() {
     fi
 
     ${SUDO} rm -rf /var/log/*pihole* &> /dev/null
-    ${SUDO} rm -rf /var/log/pihole/*pihole* &> /dev/null
-    ${SUDO} rm -rf /etc/pihole/ &> /dev/null
-    ${SUDO} rm -rf /etc/.pihole/ &> /dev/null
-    ${SUDO} rm -rf /opt/pihole/ &> /dev/null
-    ${SUDO} rm -f /usr/local/bin/pihole &> /dev/null
+    ${SUDO} rm -rf "${PI_HOLE_LOG_DIR}"/*pihole* &> /dev/null
```

Comment:
PI_HOLE_LOG_DIR, PI_HOLE_BIN_DIR, PI_HOLE_SCRIPT_DIR and PI_HOLE_CONFIG_DIR are not set within the file.

Reply:
It would be nice to have one source of truth. I think now variables are declared, generally, in `basic-install.sh` (total misnomer) and `utils.sh`. Is it possible that we put all the declarations in `basic-install.sh`? `utils.sh` would be good for helper functions but one source of truth would be nice. That install file will always be there and I don't think it would add too many more lines of code?

Reply:
I agree. I was planning to do that as a follow up PR to this one. I want to make sure I took care of all the hard-coded paths in this one first, then it will be easier to move the variables into a common declaration file after. That PR can also take care of the mixed use of `readonly` and make it explicit which places need to modify any of those values.